File: src/main/java/com/dh/ondot/schedule/api/response/ScheduleUpdateResponse.java
package com.dh.ondot.schedule.api.response;

import com.dh.ondot.core.util.DateTimeUtils;
import com.dh.ondot.schedule.domain.Schedule;

import java.time.LocalDateTime;

public record ScheduleUpdateResponse(
        Long scheduleId,
        LocalDateTime updatedAt
) {
    public static ScheduleUpdateResponse of(Schedule schedule) {
        return new ScheduleUpdateResponse(
                schedule.getId(),
                DateTimeUtils.toSeoulDateTime(schedule.getUpdatedAt())
        );
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/api/response/ScheduleUpdateResponse.kt
package com.dh.ondot.schedule.api.response

import com.dh.ondot.core.util.TimeUtils
import com.dh.ondot.schedule.domain.Schedule
import java.time.LocalDateTime

data class ScheduleUpdateResponse(
    val scheduleId: Long?,
    val updatedAt: LocalDateTime?,
) {
    companion object {
        @JvmStatic
        fun of(schedule: Schedule): ScheduleUpdateResponse {
            return ScheduleUpdateResponse(
                scheduleId = schedule.id,
                updatedAt = TimeUtils.toSeoulDateTime(schedule.updatedAt),
            )
        }
    }
}
